apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ClaimResponseResourceProvider.java

```diff
@@ -0,0 +1,107 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import ca.uhn.fhir.model.primitive.IdDt;
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.Read;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import com.codahale.metrics.MetricRegistry;
+import com.newrelic.api.agent.Trace;
+import gov.cms.bfd.server.war.commons.LoadedFilterManager;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.ClaimResponse;
+import org.hl7.fhir.r4.model.IdType;
+import org.springframework.stereotype.Component;
+
+/** This FHIR {@link IResourceProvider} adds support for R4 {@link ClaimResponse} resources. */
+@Component
+public final class R4ClaimResponseResourceProvider implements IResourceProvider {
+
+  /**
+   * A {@link Pattern} that will match the {@link ClaimResponse#getId()}s used in this application,
+   * e.g. <code>f-1234</code> or <code>m--1234</code> (for negative IDs).
+   */
+  private static final Pattern CLAIM_ID_PATTERN = Pattern.compile("(\\p{Alpha}+)-(-?\\p{Alnum}+)");
+
+  private EntityManager entityManager;
+  private MetricRegistry metricRegistry;
+  private R4SamhsaMatcher samhsaMatcher;
+  private LoadedFilterManager loadedFilterManager;
+
+  /** @param entityManager a JPA {@link EntityManager} connected to the application's database */
+  @PersistenceContext
+  public void setEntityManager(EntityManager entityManager) {
+    this.entityManager = entityManager;
+  }
+
+  /** @param metricRegistry the {@link MetricRegistry} to use */
+  @Inject
+  public void setMetricRegistry(MetricRegistry metricRegistry) {
+    this.metricRegistry = metricRegistry;
+  }
+
+  /** @param samhsaMatcher the {@link R4SamhsaMatcher} to use */
+  @Inject
+  public void setSamhsaFilterer(R4SamhsaMatcher samhsaMatcher) {
```

Comment:
I don't think we'll need a SAMHSA filter on the ClaimResponse, the procedure and diagnosis codes will only be present in the Claim